src/niftyone/figures/generator.py

```diff
@@ -0,0 +1,91 @@
+"""Generator classes for different views."""
+
+import logging
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Any, Callable, Generic, TypeVar
+
+import nibabel as nib
+import pandas as pd
+from bids2table import BIDSEntities, BIDSTable
+from matplotlib.figure import Figure
+from PIL.Image import Image
+
+import niclips.image as noimg
+
+generator_registry: dict[str, type["ViewGenerator"]] = {}
+
+T = TypeVar("T", bound="ViewGenerator")
+
+
+def register(name: str) -> Callable[[type[T]], type[T]]:
+    """Function to register generator to registry."""
+
+    def decorator(cls: type[T]) -> type[T]:
+        generator_registry[name] = cls
+        return cls
+
+    return decorator
+
+
+def create_generators(
+    config: dict[str, Any],
+) -> list["ViewGenerator"]:
+    """Function to create generators dynamically from config."""
+    generators = []
+
+    for settings in config.values():
+        query = settings.get("query", "")
+        views = settings.get("views", [])
+
+        for view in views:
+            if view in generator_registry:
+                generator_cls = generator_registry[view]
```

Comment:
I might have this be a separate function `create_generator`. It could accept common args as well as arbitrary `**kwargs` since generators may have their own options. It could also handle the error handling for unregistered generators.

Reply:
This took a little bit of hacking at, but what do you think of being able to pass the kwargs in via the config this way:

```yaml
anat:
  query: datatype == 'anat'
  views: ['three_view', 'slice_video(axis=2,vmin=0,vmax=1)']
```

If its provided, we'll parse it internally to split kwargs from the registered view. This would be an internal call within `create_generator`.

Reply:
Hm I am not sure about letting the entries in `views` be this complex. Now wouldn't building the generators require either some more involved string parsing or `eval`? What about

```
anat:
  query: datatype == 'anat'
  views:
    - name: three_view
    - name: slice_video
      axis: 2
      vmin: 0
      vmax: 1
```

Reply:
Ahh yeah, I do like this more (think I've been staring at this code too much the last few days 😅). You're right that it would require more involved string parsing otherwise.

I'll tighten this up a little bit later today with this in mind.

Reply:
What happens if the view has a Parameter `name`?

Reply:
Would `name` be associated with anything other than the view? I could see it maybe being a figure title, but I think it's more likely that any figure related kwargs use those associated kwargs (e.g. `title`, `xlabel`, `ylabel`, etc.).

Reply:
Coming back to this and the last point here. I've [implemented a more generic overlay generator](https://github.com/kaitj/niftyone/pull/2/commits/969e85c7534ac190ee4496bfa6bb860bd992e319), which assumes that the end user is generating figures with just up-to-date implementation (e.g. via config). This generator could be used to generate overlay with masks (assuming similar entities) or be used to generate overlays to check registration (e.g. fmri/dmri-to-T1w which would be using different entities).

The problem with the current config and the implementation above is that there is a high likelihood of a generated image being overwritten since the generator entities are static (e.g. they would all have the same description and extension). 

We don't have to fix it in this PR, but wanted to open the discussion (and can move it offline or to a new issue) about how to best handle this with the config.